### ISSUE TITLE
BC break in minor release 14.5

### ISSUE THREAD

**Issue by mrtnzlml** [graphql/graphql-js#2190]
Hi! I believe there is an unintentional backward compatibility break in version 14.5+ (works in 14.4). I created a repository where you can easily try it:

https://github.com/mrtnzlml/graphql-14.5-bc-break

The issue basically boils down to this query:

```graphql
query($name: String! = "MyName") {
    hello(name: $name)
}
```

In version up to 14.4 you could run such query and it would be fine. But from 14.5 GraphQL throws this error:

```text
GraphQLError: Variable "$name" of required type "String!" was not provided.
```

It's probably because I am not sending any variables BUT there is a default value so I'd not expect it to throw. To be fair, I am not sure what is the correct behavior here. I can imagine a good reasoning for both. However, it's a BC break in a minor release which is probably not intentional (?). Didn't find much about it in the changelog.

I'd personally say it's a bug because there is a `name` arg value even though it's not passed in variables.

**Comment by IvanGoncharov** [comment]
@mrtnzlml Thanks for repo with reproduce 👍 
I will take a look at it.

**Comment by IvanGoncharov** [comment]
@mrtnzlml Fixed in `14.5.7` 📦
